src/na/Architecture.cpp

```diff
@@ -545,7 +546,14 @@ auto Architecture::getPositionOffsetBy(const Point& p, const Number& rows,
     if (!nextSiteOpt.has_value()) {
       break;
     }
-    anchorSitePos = getPositionOfSite(nextSiteOpt.value());
+    const auto& newAnchorSitePos = getPositionOfSite(nextSiteOpt.value());
+    // The following check is used to decide whether the patch is located
+    // outside of the zone Patches that overlap the border of the zone are not
+    // possible with this approach yet
+    if (std::abs(newAnchorSitePos.y - anchorSitePos.y) < std::abs(dy)) {
+      break;
```

Comment:
Could it be that this line is uncovered by the existing tests?
Would be great if additional checks like that would be covered by tests so that we know they are effective.